File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/card/application/query/GetCardService.java
package com.example.accountservice.card.application.query;

import com.example.accountservice.card.domain.Card;
import com.example.accountservice.card.domain.CardException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class GetCardService {

    private final CardQuery cardQuery;

    public GetCardResult getCard(String cardId, String requesterId) {
        Card card = cardQuery.findByCardIdAndOwnerId(cardId, requesterId)
                .orElseThrow(() -> new CardException(CardException.ErrorCode.CARD_NOT_FOUND, "카드를 찾을 수 없습니다."));
        return new GetCardResult(
                card.getCardId(),
                card.getAccountId(),
                card.getOwnerId(),
                card.getBrand(),
                card.getStatus().name(),
                card.getCreatedAt());
    }
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/card/application/query/GetCardService.kt
package com.example.accountservice.card.application.query

import com.example.accountservice.card.domain.CardFindQuery
import com.example.accountservice.card.domain.CardNotFoundException
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class GetCardService(
    private val cardQuery: CardQuery,
) {
    fun getCard(
        cardId: String,
        requesterId: String,
    ): GetCardResult {
        val (cards, _) =
            cardQuery.findCards(CardFindQuery(page = 0, take = 1, cardId = cardId, ownerId = requesterId))
        val card = cards.firstOrNull() ?: throw CardNotFoundException(cardId)
        return GetCardResult(
            cardId = card.cardId,
            accountId = card.accountId,
            ownerId = card.ownerId,
            brand = card.brand,
            status = card.status.name,
            createdAt = card.createdAt,
        )
    }
}
